src/routes/movies.tsx

```diff
@@ -0,0 +1,139 @@
+import '../elements/emby-scroller/emby-scroller';
+import '../elements/emby-itemscontainer/emby-itemscontainer';
+import '../elements/emby-tabs/emby-tabs';
+import '../elements/emby-button/emby-button';
+
+import React, { FunctionComponent, useCallback, useEffect, useRef, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import * as mainTabsManager from '../components/maintabsmanager';
+import Page from '../components/Page';
+import globalize from '../scripts/globalize';
+import libraryMenu from '../scripts/libraryMenu';
+import * as userSettings from '../scripts/settings/userSettings';
+import CollectionsView from '../view/movies/CollectionsView';
+import FavoritesView from '../view/movies/FavoritesView';
+import GenresView from '../view/movies/GenresView';
+import MoviesView from '../view/movies/MoviesView';
+import SuggestionsView from '../view/movies/SuggestionsView';
+import TrailersView from '../view/movies/TrailersView';
+
+const getDefaultTabIndex = (folderId: string | null) => {
+    switch (userSettings.get('landing-' + folderId, false)) {
+        case 'suggestions':
+            return 1;
+
+        case 'favorites':
+            return 3;
+
+        case 'collections':
+            return 4;
+
+        case 'genres':
+            return 5;
+
+        default:
+            return 0;
+    }
+};
+
+const Movies: FunctionComponent = () => {
+    const [ searchParams ] = useSearchParams();
+    const currentTabIndex = parseInt(searchParams.get('tab') || getDefaultTabIndex(searchParams.get('topParentId')).toString());
+    const [ selectedIndex, setSelectedIndex ] = useState(currentTabIndex);
+    const element = useRef<HTMLDivElement>(null);
+
+    const getTabs = () => {
+        return [{
```

Comment:
you can create a const mapping outside of the component to map these indexes to their corresponding translation and Components. Then you only need to iterate that mapping or fetch the component based on the tab id.

Reply:
https://github.com/jellyfin/jellyfin-web/pull/3792/commits/2b40dbd50f39f56924b0d7ee6a3cab2dab38cbe4